src/components/twilio-voice-basic-call-control/public/twilio-voice-basic-call-control.js

```diff
@@ -22,75 +19,171 @@ class TwilioVoiceBasicCallControl extends HTMLElement {
-      <div>
-        <div id="hold-buttons">
-          <button id="hold">Hold</button>
-          <button id="resume">Resume</button>
-        </div>
-        <div id="mute-buttons">
-          <button id="mute">Mute</button>
-          <button id="unmute">Unmute</button>
-        </div>
+      <div id="basic-call-control">
+        <div id="add-participant-container" style="display: none;">
+          <input
+            type="text"
+            placeholder="Add participant"
+            id="add-participant"
+            value=""
+          />
+          <button id="add-conference">Add</button>
+        <div>
       </div>
     `;
   }
 
+  #renderCallControlButton(options) {
+    const { name, callSid, shouldHide } = options;
+    const button = document.createElement('button');
+    button.innerHTML = name;
+    button.setAttribute('id', `${callSid}-${name}`);
+    button.style.display = shouldHide ? 'none' : 'inline-block';
+    switch (name) {
+      case 'hold':
+        button.addEventListener('click', () => this.#handleHold(true, callSid));
+        break;
+      case 'resume':
+        button.addEventListener('click', () =>
+          this.#handleHold(false, callSid)
+        );
+        break;
+      case 'mute':
+        button.addEventListener('click', () => this.#handleMute(true, callSid));
+        break;
+      case 'unmute':
+        button.addEventListener('click', () =>
+          this.#handleMute(false, callSid)
+        );
+    }
+    return button;
+  }
+
+  #renderCallControlButtons() {
+    this.#removeCallControlButtons();
+    const component = this.shadowRoot.querySelector('#basic-call-control');
+    const callControlButtons = document.createElement('div');
+    callControlButtons.setAttribute('id', 'call-control-buttons');
+    component.appendChild(callControlButtons);
+
+    for (const [callSid, content] of this.#participants) {
+      const participantContainer = document.createElement('div');
+
+      const label = document.createElement('span');
+      label.innerHTML = `${content.label}: `;
+      participantContainer.appendChild(label);
+
+      const holdButton = this.#renderCallControlButton({
+        name: 'hold',
+        callSid,
+        shouldHide: content.hold,
+      });
+      participantContainer.appendChild(holdButton);
+
+      const resumeButton = this.#renderCallControlButton({
+        callSid,
+        name: 'resume',
+        shouldHide: !content.hold,
+      });
+      participantContainer.appendChild(resumeButton);
+
+      const muteButton = this.#renderCallControlButton({
+        callSid,
+        name: 'mute',
+        shouldHide: content.muted,
+      });
+      participantContainer.appendChild(muteButton);
+
+      const unMuteButton = this.#renderCallControlButton({
+        callSid,
+        name: 'unmute',
+        shouldHide: !content.muted,
+      });
+      participantContainer.appendChild(unMuteButton);
+
+      callControlButtons.appendChild(participantContainer);
+    }
+  }
```

Comment:
This is starting to get more complex. Thoughts on using a template to make the html structure more readable?

Reply:
If there's a simpler way to use templates, we should update both call control and dialer components to use it. We can add a new ticket if needed.

Reply:
I refactored to using `.innerHTML` similar to how we render other html elements in the main components. I think it is easier to read now.